frontend/src/components/core/Modal.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import { GrayTextButton } from './Button';
+
+const Content = styled.div<ContentProps>`
+  position: fixed;
+  
+  top: ${({ fullScreen }) => (fullScreen ? '10%' : '40%')};
+  height: ${({ fullScreen }) => (fullScreen ? '80%' : '150px')};
+  left: ${({ fullScreen }) => (fullScreen ? '20%' : '30%')};
+  width: ${({ fullScreen }) => (fullScreen ? '60%' : '40%')};  
+  min-height: 150px;
+
+  z-index: ${({ show }) => (show ? 2 : -2)};
+  background: ${({ theme }) => theme.colors.white};
+  overflow-y: scroll;
+  
+  border-radius: 10px;
+  box-shadow: 0 0 0 100vmax rgba(0, 0, 0, .24);
```

Comment:
What is `100vmax`?

Reply:
According to the Mozilla docs it's "equal to the larger of vw and vh" - essentially guarantees that the box shadow fills the entirety of the screen